### ISSUE TITLE
[FEATURE] Don't send notification emails when already delivered to Slack

### ISSUE THREAD

**Issue by mroy-seedbox** [PipedreamHQ/pipedream#7268]
**Is your feature request related to a problem? Please describe.**
It's awesome to receive error notifications in Slack, but I wish we wouldn't also receive the same duplicate notification via email if that notification has already been delivered to Slack. We make a point to respond to all emails & Slack notifications, so this doubles the work.

**Describe the solution you'd like**
An option to not send notifications emails if they have been successfully delivered via Slack. But if the Slack delivery fails for some reason, then we should receive the email as a backup notification.

**Do you have a workaround?**
Yes. This is just a nice-to-have/quality of life improvement.

**Additional context**
![image](https://github.com/PipedreamHQ/pipedream/assets/66686238/230fa1c3-3197-4a49-b983-6b3829c53629)

**Comment by dylburger** [comment]
@mroy-seedbox Under what conditions are you seeing duplicate error notifications? [The intent](https://pipedream.com/docs/workflows/errors/#duplicate-errors-do-not-trigger-duplicate-emails) is for this not to happen unless the same error is raised on the same workflow after another 24-hour period. I'm curious if this looks like a bug, or if you think that logic should be changed.

**Comment by dylburger** [comment]
I can repro and found out this is a known issue. The team wants to address by providing a basic UX for controlling notifications, similar to Linear:

![image (6)](https://github.com/PipedreamHQ/pipedream/assets/242668/a23c3880-5a4c-45bc-a4ce-c1ee97e70a24)

Will leave this open for tracking that.

**Comment by mroy-seedbox** [comment]
Another option could be to provide these things as triggers in the Pipedream platform. That way, customers could do whatever they want with the trigger, which would also provide a way to integrate with chat platforms other than Slack.

Zapier has a few of these triggers:
![image](https://github.com/PipedreamHQ/pipedream/assets/66686238/b515cb81-c689-4260-9258-19fa9a1bbdc8)

**Comment by dylburger** [comment]
Yes 💯 , we do plan to expose system events like this for essentially everything so that you can operate on it programmatically. We do have this for errors though! https://pipedream.com/docs/workflows/errors/#process-errors-with-custom-logic-instead-of-email
